forgerock-openbanking-uk-aspsp-common/src/main/java/com/forgerock/openbanking/common/services/openbanking/event/EventResponseUtilService.java

```diff
@@ -0,0 +1,128 @@
+/**
+ * Copyright 2019 ForgeRock AS.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.forgerock.openbanking.common.services.openbanking.event;
+
+import com.forgerock.openbanking.common.conf.discovery.ResourceLinkService;
+import com.forgerock.openbanking.common.model.openbanking.v3_0.event.FRCallbackUrl1;
+import com.forgerock.openbanking.common.model.version.OBVersion;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import uk.org.openbanking.datamodel.account.Links;
+import uk.org.openbanking.datamodel.account.Meta;
+import uk.org.openbanking.datamodel.event.*;
+
+import java.util.Collection;
+import java.util.stream.Collectors;
+
+/**
+ * Util Service class to handler the Event Notification API responses thru all versions<br />
+ */
+@Service
```

Comment:
I don't really understand why this is a service. It's a bit odd to be honest as it's name partly suggests it's a utility, but it holds state (which hopefully doesn't present a thread safety issue?).